QuantLib/ql/models/model.cpp

```diff
@@ -72,65 +76,101 @@ namespace QuantLib {
+                    const vector<Real>& w,
                     const vector<bool>& fixParameters) {
 
-        QL_REQUIRE(weights.empty() || weights.size() == instruments.size(),
-                   "mismatch between number of instruments (" <<
-                   instruments.size() << ") and weights(" <<
-                   weights.size() << ")");
+        // if uncommented test-suite fails
+        // why, oh why?
+        //QL_REQUIRE(instr.empty(), "no instruments provided");
+
+        Array prms = params();
+        if (fixParameters.empty()) {
+            fixedParameters_.resize(prms.size());
+            std::fill(fixedParameters_.begin(), fixedParameters_.end(), false);
+        } else {
+            QL_REQUIRE(fixParameters.size() == prms.size(),
+                       "mismatch between number of parametrs (" <<
+                       instr.size() << ") and fixed parameters booleans (" <<
+                       fixParameters.size() << ")");
+            fixedParameters_ = fixParameters;
+        }
+        Projection proj(prms, fixedParameters_);
+
+        if (w.empty()) {
+            weights_.resize(instr.size());
+            std::fill(weights_.begin(), weights_.end(), 1.0);
+        } else {
+            QL_REQUIRE(w.size() == instr.size(),
+                       "mismatch between number of instruments (" <<
+                       instr.size() << ") and weights (" << w.size() << ")");
+            weights_ = w;
+        }
+        CalibrationFunction f(this, instr, weights_, proj);
 
         Constraint c;
         if (additionalConstraint.empty())
             c = *constraint_;
         else
-            c = CompositeConstraint(*constraint_,additionalConstraint);
-        vector<Real> w =
-            weights.empty() ? vector<Real>(instruments.size(), 1.0): weights;
+            c = CompositeConstraint(*constraint_, additionalConstraint);
+        ProjectedConstraint pc(c, proj);
 
-        Array prms = params();
-        vector<bool> all(prms.size(), false);
-        Projection proj(prms,fixParameters.size()>0 ? fixParameters : all);
-        CalibrationFunction f(this,instruments,w,proj);
-        ProjectedConstraint pc(c,proj);
         Problem prob(f, pc, proj.project(prms));
-        shortRateEndCriteria_ = method.minimize(prob, endCriteria);
-        Array result(prob.currentValue());
-        setParams(proj.include(result));
-        Array shortRateProblemValues_ = prob.values(result);
+        endCriteria_ = method.minimize(prob, endCriteria);
+        setParams(proj.include(prob.currentValue()));
 
         notifyObservers();
     }
 
     Real CalibratedModel::value(
-                const Array& params,
-                const vector<shared_ptr<CalibrationHelper> >& instruments) {
-        vector<Real> w = vector<Real>(instruments.size(), 1.0);
+                        const Array& params,
+                        const vector<shared_ptr<CalibrationHelper> >& instr) {
```

Comment:
just to better fit into 80 colums.
I'm just changing this to helpers and h instead, to convey more meaning (and even... drum rolls... to helpers_, but just for the cpp only CalibrationFunction inner class :-) )